carbon-design-system/carbon
# feat(checkbox): add onClick prop

Closes #9577 

Adds an `onClick` handler to the `div` wrapping the checkbox `input` and its `span` label. 

Make note of the code comments I added. I'm curious if there's any additional perspectives on how to solve this issue.

#### Changelog

**New**

- checkbox: add `onClick` handler

#### Testing / Reviewing

- Open the carbon-components-react storybook
- View the Checkbox Playground story
- Click a checkbox, 3 events should fire: 
   1. `onChange`
   2. `onClick` with `event.target` indicating the `span`
   3. `onClick` with `event.target` indicating the `input`
- Click a checkbox _label_, the same set of 3 events should fire
